### PR TITLE
ci: Bump rust to 1.64.0-nightly

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Rust 1.62 has been released, let's bump the rust version to the next nightly!

## Changelog

- Build/Testing/CI

### CODE REVIEW

**Comment by Xuanwo** [comment]
@everpcpc Hi, how to resolve this error:

```shell
docker: Error response from daemon: manifest for datafuselabs/build-tool:x86_64-unknown-linux-gnu-nightly-2022-06-30 not found: manifest unknown: manifest unknown.
```

File: scripts/setup/rust-toolchain.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "nightly-2022-05-19"
+channel = "nightly-2022-06-30"
```
Comment (PsiACE) [inline scripts/setup/rust-toolchain.toml:2]: I think this should be a separate PR and expect it to be merged first, to build the build-tool image.

Comment (Xuanwo) [reply scripts/setup/rust-toolchain.toml:2]: Only update `scripts/setup/rust-toolchain.toml`?

Comment (PsiACE) [reply scripts/setup/rust-toolchain.toml:2]: https://databend.rs/doc/faq/routine-maintenance

**Comment by Xuanwo** [comment]
Depends on https://github.com/datafuselabs/databend/pull/6376

File: scripts/setup/rust-toolchain.toml (hunk 1 above)
Comment (Xuanwo) [reply scripts/setup/rust-toolchain.toml:2]: This workflow seems broken.

`build-tools` also requires changes in `rust-toolchain.toml`.

Comment (Xuanwo) [reply scripts/setup/rust-toolchain.toml:2]: @everpcpc for help

**Comment by Xuanwo** [comment]
@PsiACE @everpcpc How about get this PR merged first and than fix possible other errors?

**Comment by Xuanwo** [comment]
Reverted for more discussion.

**Comment by Xuanwo** [comment]
Maybe we need to build new image by hand first?

**Comment by Xuanwo** [comment]
Try https://github.com/datafuselabs/databend/pull/6378

**Comment by Xuanwo** [comment]
@mergify update

**Comment by Xuanwo** [comment]
All tests passed without regression. Thanks rust community!

**Review by PsiACE — APPROVED** [review]

**Review by everpcpc — APPROVED** [review]
